src/workerd/io/actor-sqlite-test.c++

```diff
@@ -930,21 +930,48 @@ KJ_TEST("calling deleteAll() preserves alarm state if alarm is set") {
   test.pollAndExpectCalls({});
   KJ_ASSERT(expectSync(test.getAlarm()) == oneMs);
 
-  KJ_ASSERT(!test.actor.isCommitScheduled());
-  ActorCache::DeleteAllResults results = test.actor.deleteAll({});
-  KJ_ASSERT(test.actor.isCommitScheduled());
-  KJ_ASSERT(results.backpressure == kj::none);
-  KJ_ASSERT(expectSync(test.getAlarm()) == oneMs);
+  {
+    KJ_ASSERT(!test.actor.isCommitScheduled());
+    ActorCache::DeleteAllResults results = test.actor.deleteAll({});
+    KJ_ASSERT(test.actor.isCommitScheduled());
+    KJ_ASSERT(results.backpressure == kj::none);
+    KJ_ASSERT(expectSync(test.getAlarm()) == oneMs);
 
-  auto commitFulfiller = kj::mv(test.pollAndExpectCalls({"commit"})[0]);
-  KJ_ASSERT(results.count.wait(test.ws) == 0);
-  KJ_ASSERT(expectSync(test.getAlarm()) == oneMs);
+    auto commitFulfiller = kj::mv(test.pollAndExpectCalls({"commit"})[0]);
```

Comment:
For my own education, we only expect to see `scheduleRun(1ms)` when setAlarm is called, and not on subsequent resets?  Presumably, that's because the scheduled run gets scheduled outside of this deletion?

Reply:
Pretty much, yeah.  The `scheduleRun()` hook is called by ActorSqlite as part of the committing logic -- implemented in `startPrecommitAlarmScheduling()` and `commitImpl()` -- to request a change to the alarm manager's scheduled alarm notification time.  If ActorSqlite knows the currently scheduled alarm doesn't need to be updated, it doesn't call `scheduleRun()`.

But thinking about this a bit more, I guess the point of confusion might be why we don't see a `scheduleRun(kj::none)` in the test corresponding to the `db.reset()` and a subsequent `scheduleRun(1ms)` corresponding to the alarm state reset? That's because we don't run the committing logic for the `db.reset()`, only for the state reset.  And because we're restoring the original state, the committing logic sees no change, so `scheduleRun()` is not called.

I haven't yet implemented Kenton's suggestion of simplifying the deleteAll() logic by performing a redundant commit, but I'm pretty sure this won't change the situation, since the promise that deleteAll() queues calls `commitCallback()` directly, rather than through the alarm commit logic.  And I'm pretty sure that's OK?  But might need to think about it a bit more to be sure.